pkg/model/config/config.go

```diff
@@ -78,39 +81,105 @@ func (c Config) IsV3() bool {
 	return c.Version == Version3Alpha
 }
 
-// HasResource returns true if API resource is already tracked
-func (c Config) HasResource(target GVK) bool {
-	// Return true if the target resource is found in the tracked resources
+type unsupportedFieldError struct {
+	fieldName string
+	version   string
+}
+
+func (e unsupportedFieldError) Error() string {
+	return fmt.Sprintf("`%s` field found for %s", e.fieldName, e.version)
+}
+
+func (c Config) errorForField(fieldName string) error {
+	return unsupportedFieldError{fieldName: fieldName, version: c.Version}
+}
+
+// validate checks if a Config is valid or it has some fields that shouldn't be present for that version
+func (c Config) validate() error {
+	if c.IsV2() {
+		if c.ProjectName != "" {
+			return c.errorForField("projectName")
+		}
+		for _, r := range c.Resources {
+			if r.Domain != "" {
+				return c.errorForField("resources[].domain")
+			}
+			if r.Plural != "" {
+				return c.errorForField("resources[].plural")
+			}
+			if r.Path != "" {
+				return c.errorForField("resources[].plural")
+			}
```

Comment:
As we spoke the name Endpoint shows more appropriated than PATH. So, could you change it.